### DISCUSSION TITLE
Using Maven artifacts in PDE/RCP in a Tycho build doesn't consider repositories declared in pom.xml

### DISCUSSION THREAD

**Discussion by khatchad** [eclipse-tycho/tycho#2322] [Q&A] [answered]
I am using Maven artifacts in PDE/RCP in a Tycho builds (see [this article](https://xn--lubisoft-0za.gmbh/en/articles/using-maven-artifacts-in-pde-rcp-and-tycho-builds/)) in a multi-module Maven project with eclipse packaging. I followed the instructions in the article, but M2E doesn't seem to consider repositories declared in the main `pom.xml` file. Specifically, my Maven dependency is hosted by jitpack.io. The command line build works fine, but in order to Eclipse to pick it up, I need to copy the repository information to my `settings.xml` in my home directory. Only then does it find the dependency.

Is this the expected behavior?

**Comment by laeubi** [comment, marked as answer]
> Is this the expected behavior?

Short answer, yes.

Long answer is:

1. A target is not connected to any pom (even though you can declare dependencies like a pom), so it can't use any repositories declared there.
2. You can define additional in the settings of the target location:
![grafik](https://user-images.githubusercontent.com/1331477/229872266-a3d21500-8766-4ecc-b05a-52dd68f13288.png)

**Reply by khatchad** [reply]
Ah, I didn't realize that "additional artifact repositories" meant ones that were already specified in the `pom.xml`.

**Reply by laeubi** [reply]
As said forget about the `pom.xml` it is not related to the target file.

**Reply by khatchad** [reply]
Isn't an artifact repository kind of like a "software site?" Shouldn't that somehow be part of the target file somehow?

**Reply by khatchad** [reply]
NM, I see that if you add it using the UI, it becomes part of the target file.

**Reply by laeubi** [reply]
Yes it will be stored in the target file, if the repository requires authentication, you still need to add it to the settings.xml!
